mycpp/pass_state.py

```diff
@@ -538,8 +544,16 @@ def DumpControlFlowGraphs(cfgs: dict[str, ControlFlowGraph],
     directory as text files that can be consumed by datalog.
     """
     edge_facts = '{}/cf_edge.facts'.format(facts_dir)
-    fact_files = {}
+
     os.makedirs(facts_dir, exist_ok=True)
+    # Open files for all facts that we might emit even if we don't end up having
+    # anything to write to them. Souffle will complain if it can't find the file
+    # for anything marked as an input.
+    fact_files = {
+        fact_type.name():
+        open('{}/{}.facts'.format(facts_dir, fact_type.name()), 'w')
+        for fact_type in Fact.__subclasses__()
```

Comment:
This feels like a lot of indirection for what is just a bunch of strings ... I think it could be just a list of strings:

```
fact_subdirs = ['assign', 'bind', ...]
```

And then I think you could write `assert len(fact_subdirs) == Fact.__subclasses__()` if really necessary

I try to avoid reflection like `__subclasses__` in the main code -- since those idioms don't exist in say C++

Reply:
I dunno. There's zero risk of this file getting translated to C++, right? I think it would be much more (relatively speaking lol) annoying to have to update this list every time we add a new fact. If we'd be using `__subclasses__()` in the assert to remind ourselves to come update the list, why not just use it directly?

Reply:
Well I still think we should move mycpp to use our own ASDL  schema, "collapsing" the MyPy AST a bit

In that case, it won't be that far from C++

And mycpp is too slow, so there is some motivation for that :)